ui/src/time_picker.tsx

```diff
@@ -212,8 +213,8 @@ export const
                 value={value}
                 label={label}
                 open={isDialogOpen}
-                onChange={value => setValue(value as Date)}
-                onAccept={value => onSelectTime(value as Date)}
+                onChange={value => onChangeTime(value as Date)}
```

Comment:
Let's use proper types here.

```suggestion
                onChange={onChangeTime}
```

Reply:
What do you mean by proper types? onChange gives us the value of type `unknown`, which is basically a Date object every time. Do you mean I should replace the param type in my `onChangeTime` handler with `unknown`? 

```ts
- onChangeTime = (time: D | null) => {
+ onChangeTime = (time: unknown) => {
```

Reply:
>onChange gives us the value of type unknown, which is basically a Date object every time

If it were true, it wouldn't be typed as unknown, but Date.

>Do you mean I should replace the param type in my onChangeTime handler with unknown?

The current `onChangeTime` function signature doesn't match the one in the lib. In addition to the value, there is also `keyboardInputValue` which I assume allows for selecting time with a keyboard as well so would be good to support that also.

Reply:
> I assume allows for selecting time with a keyboard as well so would be good to support that also

IIRC the keyboard input was supported during development of this component but in later phases we agreed on removing the support. I can't remember the specific reason, but I think there were a lot of edge cases to handle and the code was too robust. Also we wanted to make it consistent with a date picker where you cannot use keyboard input as well. You can pick only what the picker offers you to select and also you don't need to care about formats (should I write `04:00`, `4:00`, `4:00 am` or `4:00am`?).

However, If you want to support it, I can create a feature request and/or add there a TODO.

Reply:
Ah, I remember now, fair point. Let's leave it as is then and handle only the `unknown` part of my prev comment.